torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -3739,3 +3740,235 @@ def test_tensorpipe_options_throw_on_timedelta_timeout(self):
+        options = self.rpc_backend_options
+        options.set_map_location("none_exist", {-1: 0})
+        self._test_map_locations(options, "Wrong worker names")
+
+    def test_map_locations_invalid_max_local_device(self):
+        options = self.rpc_backend_options
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options.set_map_location(dst, {torch.cuda.device_count(): -1})
+
+        self._test_map_locations(options)
+
+    @skip_if_lt_x_gpu(1)
+    def test_map_locations_invalid_min_local_device(self):
+        options = self.rpc_backend_options
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options.set_map_location(dst, {-2: 0})
+
+        self._test_map_locations(options)
+
+    @skip_if_lt_x_gpu(1)
+    def test_map_locations_invalid_min_remote_device(self):
+        options = self.rpc_backend_options
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options.set_map_location(dst, {0: -2})
+
+        self._test_map_locations(options)
+
+    def test_map_locations_invalid_max_remote_device(self):
+        options = self.rpc_backend_options
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options.set_map_location(dst, {-1: torch.cuda.device_count()})
+
+        self._test_map_locations(options)
+
+    @skip_if_lt_x_gpu(1)
+    def test_map_locations_many_to_one(self):
+        options = self.rpc_backend_options
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options.set_map_location(dst, {-1: 0})
+        options.set_map_location(dst, {0: 0})
+
+        self._test_map_locations(options)
+
+    @skip_if_lt_x_gpu(1)
+    def test_map_locations_ambiguous_cpu(self):
+        options = self.rpc_backend_options
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options.set_map_location(dst, {0: -1})
+
+        self._test_map_locations(options, "device mapping for CPU.*ambiguous")
+
+    def test_map_locations_cpu(self):
+        options = self.rpc_backend_options
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options.set_map_location(dst, {-1: -1})
+
+        rpc.init_rpc(
+            name=worker_name(self.rank),
+            backend=self.rpc_backend,
+            rank=self.rank,
+            world_size=self.world_size,
+            rpc_backend_options=options,
+        )
+
+        ret = rpc.rpc_sync(dst, torch.add, args=(torch.zeros(2), torch.ones(2)))
+        self.assertEqual(ret, torch.zeros(2) + torch.ones(2))
+
+        rpc.shutdown()
+
+    @staticmethod
+    def _gpu_add(x, y):
+        if all([x.is_cuda, x.device.index == 0, y.is_cuda, y.device.index == 0]):
+            return (x + y).to(1)
+        else:
+            raise ValueError("Wrong device affinity")
+
+    @skip_if_lt_x_gpu(2)
+    def test_map_locations_gpu(self):
+        options = self.rpc_backend_options
+        dst = worker_name((self.rank + 1) % self.world_size)
+        options.set_map_location(dst, {-1: 0})
```

Comment:
> Shouldn't this test fail because of the invalid CPU mapping?

Do you mean we should make this fail by disallowing CPU mapping?

Reply:
I thought that CPUs could only be mapped to CPUs and thus I interpreted this mapping as invalid, but I then realized that we weren't aligned on how CPUs are to be mapped. Since, as I understand it, we're going to disallow CPUs in the device map for now, this will get resolved as part of that.